sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -27,8 +27,7 @@ public partial class ArmClient
         private bool? _canUseTagResource;
 
         internal virtual Dictionary<ResourceType, string> ApiVersionOverrides { get; } = new Dictionary<ResourceType, string>();
-
-        internal ConcurrentDictionary<string, Dictionary<string, string>> ResourceApiVersionCache { get; } = new ConcurrentDictionary<string, Dictionary<string, string>>();
+        internal ConcurrentDictionary<string, Dictionary<string, string>> ResourceApiVersionCache { get; } = new ConcurrentDictionary<string, Dictionary<string, string>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This ConcurrentDictionary has a ignore case comparer - do we need a ignore case comparer in the Dictionary as value? Sometimes the casing of the resource types is not consistent and I believe it is better we could make this thing more robust

Reply:
Good point. I've made the changes and updated test cases.

Reply:
If the reason for this is that the profile files sometimes have mis-cased resource types I don't think this is the right place to normalize as it opens up possibilities for mis matches in other areas that may not be resilient to it.  These are used in URIs which are case sensitive by design.

Instead the api that reads in the profile could normalize to the real resource type if possible OR we may need to fix the profiles.  Given that these were never used in a way that required them to be correct before it makes sense there might be a few mistakes in there that need to be corrected.

Reply:
The `Equals` method of `ResourceType` is using ignore-case comparison, here the dictionary keys use the split namespace and type from ResourceType correspondingly, therefore I think adding the `OrdinalIgnoreCase` is consistent with the `Equals` method of `ResourceType`.  As a comparison, `ApiVersionOverrides` uses `ResourceType` as the key, and when invoking `TryGetValue(key)`, the underlying resource type string for the key is case-insensitive.
https://github.com/Azure/azure-sdk-for-net/blob/3bc773a598f2a358bd0147daf4609dd0e47bc909/sdk/core/Azure.Core/src/ResourceType.cs#L120-L123